### ISSUE TITLE
type for createAnimatedComponent param is vague (any)

### ISSUE THREAD

**Issue by goleary** [software-mansion/react-native-reanimated#1537]
## Description

When I call `Reanaimated.createAnimatedComponent()` with a functional component I get this error at runtime.

```
`createAnimatedComponent` does not support stateless functional components; use a class component instead.
```
Here's an example snack: https://snack.expo.io/@goleary/reanimated.createanimatedcompoent-type-failure
## Steps To Reproduce

call `Reanimated.createAnimatedComponen()` & pass it a functional component.

### Expected behavior

It would be awesome if the type of the param that this function accepts would help the user understand this code will fail at runtime because whatever is being passed won't work (is not a class component).

### Actual behavior
createAnimatedComponent accepts anything passed to it and fails at runtime.

## Snack or minimal code example
https://snack.expo.io/@goleary/reanimated.createanimatedcompoent-type-failure

## Package versions

- React: 16.13.1
- React Native: 0.63.2
- React Native Reanimated: 1.13.0

**Comment by neiker** [comment]
Related: https://github.com/software-mansion/react-native-reanimated/discussions/1527

**Comment by jakub-gonet** [comment]
#1466 resolved the issue with the any, #1539 should close this one.

**Comment by WoLewicki** [comment]
Looking at the PR linked in #1539, it seems that this issue should be resolved by https://github.com/software-mansion/react-native-reanimated/pull/1590. I will close this issue then. Feel free to comment here if I am wrong and we can reopen it then.
